### PR TITLE
add failpoint test for db operation

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
add some failpoint about db operation, and add integration test
issue: https://internal.pingcap.net/jira/browse/TOOL-1394

### What is changed and how it works?
1. add some fail point about db operation
2. use these fail point to add integration test
- initial unit failed, and find a bug which will lead to panic on dm-worker and dm-master
- start task failed
- relay log failed

### Check List 

Tests 
 - Integration test

### CODE REVIEW

File: dm/worker/subtask.go [hunk 1]
```diff
@@ -116,7 +117,11 @@ func (st *SubTask) Init() error {
 			u.Close()
 		}
 
-		st.initialized.Set(true)
+		if len(needCloseUnits) == 0 {
+			st.initialized.Set(true)
+		} else {
+			st.initialized.Set(false)
+		}
```
Comment (kennytm) [inline dm/worker/subtask.go]:
```go
st.initialized.Set(len(needCloseUnits) == 0)
```

**Comment by WangXiangUSTC** [comment]
@amyangfei have already add new test in others_integration.txt

File: dm/worker/subtask.go (hunk 1 above)
Comment (WangXiangUSTC) [reply dm/worker/subtask.go]: done

File: loader/db.go [hunk 2]
```diff
@@ -131,6 +134,13 @@ func (conn *Conn) executeSQLCustomRetry(ctx *tcontext.Context, sqls []string, en
 
 		startTime := time.Now()
 		err = executeSQLImp(ctx, conn.db, sqls)
+
+		failpoint.Inject("LoadExecCreateTableFailed", func() {
+			if i == 0 && len(sqls) == 1 && strings.Contains(sqls[0], "CREATE TABLE") {
+				err = domain.ErrInfoSchemaChanged
```
Comment (GregoryIan) [inline loader/db.go]: why only inject `ErrInfoSchemaChanged ` error here? can we support to inject custom error?

File: tests/_utils/run_dm_ctl [hunk 3]
```diff
@@ -26,7 +27,6 @@ for ((i=1; i<$#; i+=2)); do
     got=$(sed "s/$value/$value\n/g" $dmctl_log | grep -c "$value")
     if [ "$got" != "$expected" ]; then
         echo "command: $cmd $value count: $got != expected: $expected"
-        cat $dmctl_log
```
Comment (GregoryIan) [inline tests/_utils/run_dm_ctl]: why remove it

File: tests/load_interrupt/run.sh [hunk 4]
```diff
@@ -59,7 +59,8 @@ function run() {
     check_row_count 1
     check_row_count 2
 
-    export GO_FAILPOINTS=''
+    # only failed at the first time, will retry later and success
+    export GO_FAILPOINTS='github.com/pingcap/dm/loader/LoadExecCreateTableFailed=return(true)'
```
Comment (GregoryIan) [inline tests/load_interrupt/run.sh]: how to check it?

**Comment by GregoryIan** [comment]
Good Job！

File: loader/db.go (hunk 2 above)
Comment (WangXiangUSTC) [reply loader/db.go]: use error code, https://github.com/pingcap/dm/pull/206/commits/3b3579559aa458333d25826569faea6fcd36d828

File: tests/load_interrupt/run.sh (hunk 4 above)
Comment (WangXiangUSTC) [reply tests/load_interrupt/run.sh]: add log check
```
 # LoadExecCreateTableFailed error return twice
 err_cnt=`grep LoadExecCreateTableFailed $WORK_DIR/worker1/log/dm-worker.log | wc -l`
 if [ $err_cnt -ne 2 ]; then
     echo "error LoadExecCreateTableFailed's count is not 2"
     exit 2
 fi
```

**Comment by WangXiangUSTC** [comment]
@GregoryIan @amyangfei PTAL again

File: loader/db.go [hunk 5]
```diff
@@ -131,6 +134,25 @@ func (conn *Conn) executeSQLCustomRetry(ctx *tcontext.Context, sqls []string, en
...
+			}
+
+			errCode, err1 := strconv.ParseUint(items[0], 16, 16)
+			errNum, err2 := strconv.ParseInt(items[1], 16, 16)
+			if err1 != nil || err2 != nil {
+				ctx.L().Fatal("failpoint LoadExecCreateTableFailed's value is invalid", zap.String("val", val.(string)), zap.Strings("items", items), zap.Error(err1), zap.Error(err2))
+			}
+
+			if i < int(errNum) && len(sqls) == 1 && strings.Contains(sqls[0], "CREATE TABLE") {
+				err = tmysql.NewErr(uint16(errCode))
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply loader/db.go]: but it is difficult to know the arguments' type, some argument is int type, others may be string type. and we only need concern  the error code in our code

File: tests/_utils/test_prepare [hunk 6]
```diff
@@ -25,6 +25,15 @@ function cleanup2() {
     pkill -hup dm-tracer.test 2>/dev/null || true
 }
 
+function cleanup() {
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply tests/_utils/test_prepare]: update in https://github.com/pingcap/dm/pull/206/commits/9d7936a1b1e7a85f716667e064f43fd29e8c444f

File: loader/db.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (amyangfei) [reply loader/db.go]: ok

File: loader/db.go [hunk 7]
```diff
@@ -131,6 +134,25 @@ func (conn *Conn) executeSQLCustomRetry(ctx *tcontext.Context, sqls []string, en
...
 		err = executeSQLImp(ctx, conn.db, sqls)
+
+		failpoint.Inject("LoadExecCreateTableFailed", func(val failpoint.Value) {
+			items := strings.Split(val.(string), ",")
+			if len(items) != 2 {
+				ctx.L().Fatal("failpoint LoadExecCreateTableFailed's value is invalid", zap.String("val", val.(string)))
+			}
+
+			errCode, err1 := strconv.ParseUint(items[0], 16, 16)
+			errNum, err2 := strconv.ParseInt(items[1], 16, 16)
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply loader/db.go]: done

**Comment by WangXiangUSTC** [comment]
@GregoryIan PTAL again

**Pull request merged**